test/prob/test_fixture_ccdf_log.hpp

```diff
@@ -257,8 +260,8 @@ class AgradCcdfLogTestFixture : public ::testing::Test {
   // works for <var>
   double calculate_gradients_1storder(vector<double>& grad, var& ccdf_log,
                                       vector<var>& x) {
+    stan::math::set_zero_all_adjoints();
```

Comment:
These gradient functions get called a lot in a sequence. If we do `stan::math::recover_memory` we clear the autodiff stack and then the tests aren't meaningful. I switched the `recover_memory` s to `set_zero_all_adjoint` s and put `recover_memory` calls in the tests that use the `calculate_gradients_*` functions.